tests/integration/operator/switch_cr_test.go

```diff
@@ -0,0 +1,244 @@
+			}
+
+			checkControllerInstallation(t, ctx, istioCtl, workDir, path.Join(ManifestTestDataPath, "all_on.yaml"))
+			checkControllerInstallation(t, ctx, istioCtl, workDir, path.Join(ProfilesPath, "default.yaml"))
+			checkControllerInstallation(t, ctx, istioCtl, workDir, path.Join(ProfilesPath, "demo.yaml"))
+		})
+}
+
+// checkInstallStatus check the status of IstioOperator CR from the cluster
+func checkInstallStatus(cs kube.Cluster) error {
+	gvr := schema.GroupVersionResource{
+		Group:    "install.istio.io",
+		Version:  "v1alpha1",
+		Resource: "istiooperators",
+	}
+	us, err := cs.GetUnstructured(gvr, "istio-system", "test-istiocontrolplane")
+	if err != nil {
+		return fmt.Errorf("failed to get istioOperator resource: %v", err)
+	}
+	usIOPStatus := us.UnstructuredContent()["status"].(map[string]interface{})
+	iopStatusString, err := json.Marshal(usIOPStatus)
+	if err != nil {
+		return fmt.Errorf("failed to marshal istioOperator status: %v", err)
+	}
+	status := &api.InstallStatus{}
+	jspb := jsonpb.Unmarshaler{AllowUnknownFields: true}
+	if err := jspb.Unmarshal(bytes.NewReader(iopStatusString), status); err != nil {
+		return fmt.Errorf("failed to unmarshal istioOperator status: %v", err)
+	}
+	if status.Status != api.InstallStatus_HEALTHY {
+		return fmt.Errorf("expect IstioOperator status to be healthy, but got: %v", status.Status)
+	}
+	var errs util.Errors
+	for cn, cnstatus := range status.ComponentStatus {
+		if cnstatus.Status != api.InstallStatus_HEALTHY {
+			errs = util.AppendErr(errs, fmt.Errorf("expect component: %s status to be healthy,"+
+				" but got: %v", cn, cnstatus.Status))
+		}
+	}
+
+	return errs.ToError()
+}
+
+func checkControllerInstallation(t *testing.T, ctx framework.TestContext, istioCtl istioctl.Instance, workDir string, iopFile string) {
+	scopes.CI.Infof(fmt.Sprintf("=== Checking istio installation by operator controller with iop file: %s===\n", iopFile))
+	s, err := image.SettingsFromCommandLine()
+	if err != nil {
+		t.Fatal(err)
+	}
+	iop, err := ioutil.ReadFile(iopFile)
+	if err != nil {
+		t.Fatalf("failed to read iop file: %v", err)
+	}
+	metadataYAML := `
+metadata:
+  name: test-istiocontrolplane
+  namespace: istio-system
+`
+	iopcr, err := util.OverlayYAML(string(iop), metadataYAML)
+	if err != nil {
+		t.Fatalf("failed to overlay iop with metadata: %v", err)
+	}
+	iopCRFile := filepath.Join(workDir, "iop_cr.yaml")
+	if err := ioutil.WriteFile(iopCRFile, []byte(iopcr), os.ModePerm); err != nil {
+		t.Fatalf("failed to write iop cr file: %v", err)
+	}
+	initCmd := []string{
+		"operator", "init",
+		"--wait",
+		"-f", iopCRFile,
+		"--hub=" + s.Hub,
+		"--tag=" + s.Tag,
+	}
+	// install istio using operator controller
+	istioCtl.InvokeOrFail(t, initCmd)
+	cs := ctx.Environment().(*kube.Environment).KubeClusters[0]
+
+	// takes time for reconciliation to be done
+	scopes.CI.Infof("waiting for reconciliation to be done")
+	time.Sleep(60 * time.Second)
```

Comment:
can't we poll the status of the IOP resource instead of sleeping? we're updating it when reconciliation is finished

Reply:
+1, please don't sleep in tests

Reply:
done